### ISSUE TITLE
HoveredIdTimer doesn't increment on disabled items

### ISSUE THREAD

**Issue by s9w** [ocornut/imgui#3419]
**Version/Branch of Dear ImGui:**
Version: 1.77
Branch: master

**Back-end/Renderer/Compiler/OS**
Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Operating System: Win10

**My Issue/Question:**
For #1940, there was the new `ImGuiHoveredFlags_AllowWhenDisabled` added. That works very well to allow feedback on disabled buttons etc. Unrelated to that, in #1485 there was a way posted to implement delayed tooltips by using `GImGui->HoveredIdTimer`. However that `HoveredIdTimer` does not increment with disabled Buttons.

I'm not 100% sure that it is supposed to, although it seems reasonable.

**Standalone, minimal, complete and verifiable example:**
```c++
ImGui::PushItemFlag(ImGuiItemFlags_Disabled, true);
ImGui::Button("disabled!");
ImGui::PopItemFlag();

if(ImGui::IsItemHovered(ImGuiHoveredFlags_AllowWhenDisabled) &&
   GImGui->HoveredIdTimer >= delay)
{
   // GImGui->HoveredIdTimer is always 0.0, hence this only works with delay==0.0
}
```

**Comment by ocornut** [comment]
Hello,

FYI i have renamed the topic has this technically doesn't have anything to do with `ImGuiHoveredFlags_AllowWhenDisabled`, it's generally that `g.HoveredID` is not set on disabled item and therefore `g.HoveredIdTimer` won't increment. 

But somehow the way `IsItemHovered()` work will work on disabled item - as `IsItemHovered()` works on items that don't have ID, it's pulling the last item data which is filled regardless of the value of `g.HoveredId` and disabled state.

I think we should change the code to fill `g.HoveredId` on disabled item but this may have a few side effects so need to be done carefully. In fact we have a branch by @rokups that attempted to do that already (initially for a different reason). I will try to review their work on that.

**Comment by ocornut** [comment]
This is now fixed by 5dd1e38b7a5a6ed6c6d94c54c6eb54ba3a88ed81.
